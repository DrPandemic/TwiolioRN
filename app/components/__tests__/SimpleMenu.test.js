// @flow

import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../../store');

import { PSimpleMenu } from '../SimpleMenu';

test('can go back', () => {
  const spy = jest.fn();
  const menu = new PSimpleMenu({
    goBack: spy,
    location: { state: 'foo' },
  });

  const renderedButton = menu.renderButton();
  renderedButton.props.onPress();

  expect(spy).toBeCalled();
});

test('renders a menu with an id', () => {
  const list = renderer.create(
    <PSimpleMenu
      goBack={ () => {} }
      location={{ state: 'foo' }}
    >
    </PSimpleMenu>
  ).toJSON();

  expect(list).toMatchSnapshot();
});

test('renders a menu without an id', () => {
  const list = renderer.create(
    <PSimpleMenu
      goBack={ () => {} }
      location={{ state: { conversationId: 'foo' } }}
    >
    </PSimpleMenu>
  ).toJSON();

  expect(list).toMatchSnapshot();
});
