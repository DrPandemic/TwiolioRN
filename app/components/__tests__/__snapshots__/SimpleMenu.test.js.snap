// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a menu with an id 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 25,
        },
        Object {},
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 20,
        "marginLeft": 10,
      }
    }
  />
</View>
`;

exports[`renders a menu without an id 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 25,
        },
        Object {},
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 20,
        "marginLeft": 10,
      }
    }
  />
</View>
`;
